src/lx16a-servo.cpp

```diff
@@ -4,96 +4,145 @@
 // write a command with the provided parameters
 // returns true if the command was written without conflict onto the bus
 bool LX16AServo::write(uint8_t cmd, const uint8_t *params, int param_cnt) {
-    if (param_cnt < 0 || param_cnt > 4) return false;
-    // prepare packet in a buffer
-    int buflen = 6 + param_cnt;
-    uint8_t buf[buflen];
-    buf[0] = 0x55;
-    buf[1] = 0x55;
-    buf[2] = _id;
-    buf[3] = buflen-3;
-    buf[4] = cmd;
-    for (int i=0; i<param_cnt; i++) buf[5+i] = params[i];
-    uint8_t cksum = 0;
-    for (int i=2; i<buflen-1; i++) cksum += buf[i];
-    buf[buflen-1] = ~cksum;
+	if (param_cnt < 0 || param_cnt > 4)
+		return false;
```

Comment:
Quick comment here: I'm OK with
```
if (condition) statement;
```
or
```
if (condition) {
    statement;
}
```
but not the multiline version without braces 'cause it's too easy in a future edit to make a mistake.